### PR TITLE
Observe query updates

### PR DESCRIPTION
**Pull request by poojamat**
*Issue #, if available:*

*Description of changes:*
Added subscription cancel on sqlProcessor error for observe query. Removed batching after sync for observe query.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: aws-datastore/src/androidTest/java/com/amplifyframework/datastore/storage/sqlite/SQLiteStorageAdapterObserveQueryTest.java [hunk 1]
```diff
@@ -609,7 +608,7 @@ public void queryWithOrderByRelatedModel() throws DataStoreException, Interrupte
      * @throws DataStoreException   On unexpected failure manipulating items in/out of DataStore
      * @throws InterruptedException On unexpected failure manipulating items in/out of DataStore
      */
-    @Ignore("Failing in build")
+    //@Ignore("Failing in build")
```
Comment (rjuliano) [inline aws-datastore/src/androidTest/java/com/amplifyframework/datastore/storage/sqlite/SQLiteStorageAdapterObserveQueryTest.java:611]: You can probably just remove this.

**Review by rjuliano — APPROVED** [review]
One minor comment, but other than that LGTM!!!

File: aws-datastore/src/androidTest/java/com/amplifyframework/datastore/storage/sqlite/SQLiteStorageAdapterObserveQueryTest.java (hunk 1 above)
Comment (poojamat) [reply aws-datastore/src/androidTest/java/com/amplifyframework/datastore/storage/sqlite/SQLiteStorageAdapterObserveQueryTest.java]: Done.

**Comment by poojamat** [comment]
feat: Updates to observe query API to not batch the changes after Sync and cancel subscription in some error scenarios.

**Pull request merged**